test/server/cards/05_LOF/leaders/ReyNobody.spec.ts

```diff
@@ -0,0 +1,174 @@
+
+describe('Rey, Nobody', function() {
+    integration(function(contextRef) {
+        describe('Rey\'s undeployed ability', function() {
+            beforeEach(function () {
+                return contextRef.setupTestAsync({
+                    phase: 'action',
+                    player1: {
+                        hand: ['drain-essence', 'bolstered-endurance', 'yoda#old-master'],
+                        groundArena: ['guardian-of-the-whills', 'atst'],
+                        leader: 'rey#nobody',
+                        resources: 6 // making Rey undeployable makes testing the activated ability's condition smoother
+                    },
+                    player2: {
+                        hand: ['force-choke', 'daughter-of-dathomir'],
+                        spaceArena: ['alliance-xwing'],
+                        leader: { card: 'luke-skywalker#faithful-friend', deployed: true }
+                    }
+                });
+            });
+
+            it('should only have an effect if the controller played a Force non-unit card this phase, but still be usable otherwise', function () {
+                const { context } = contextRef;
+
+                // No Force card played; ability has no effect
+                context.player1.clickCard(context.rey);
+                context.player1.clickPrompt('Use it anyway');
+
+                expect(context.rey.exhausted).toBe(true);
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should damage a unit if a Force event card was played by controller', function () {
+                const { context } = contextRef;
+
+                // Play a Force event card
+                context.player1.clickCard(context.drainEssence);
+                context.player1.clickCard(context.guardianOfTheWhills);
+                context.player2.passAction();
+
+                // Use ability with effect
+                context.player1.clickCard(context.rey);
+
+                // Select a unit to damage
+                expect(context.player1).toBeAbleToSelectExactly([context.allianceXwing, context.lukeSkywalker, context.atst]);
+                context.player1.clickCard(context.atst);
+
+                // Check board state
+                expect(context.rey.exhausted).toBe(true);
+                expect(context.atst.damage).toBe(1);
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should damage a unit if a Force upgrade card was played by controller', function () {
```

Comment:
Rey's ability should also resolve if the player played a Force pilot as an upgrade (e.g. Luke Skywalker, You Still With Me?). I think for that to work, the ability logic should incorporate the `playedAsType` into the cards played watcher.